### DISCUSSION TITLE
Google cloud https

### DISCUSSION THREAD

**Discussion by aflextr** [coder/code-server#3907] [Q&A]
How can I make my virtual machine https domain in google cloud?

Note: "code-server --link" not working.

**Comment by jsjoeio** [comment]
Hmm...did you follow [this part of the docs](https://coder.com/docs/code-server/v3.11.0/requirements#set-up-a-vm-on-google-cloud)?

I assumed it would cover that, but haven't gone through it myself. Let me know! (We may need to update that)

**Reply by aflextr** [reply]
@jsjoeio yes, I did the same but https didn't happen yet.

**Reply by jsjoeio** [reply]
Hmm...did you try following [this](https://tech.jocodoma.com/2019/06/11/Enable-HTTPS-and-Obtain-SSL-TLS-Certificate-for-Google-Cloud-VM/)? Let me know if that works and if it does, I'll update our docs

**Reply by aflextr** [reply]
@jsjoeio hmm, after some research I found out that google cloud doesn't have its own public fqdn. To make Google VMs https, it was necessary to add an external domain.

**Reply by jsjoeio** [reply]
> fqdn

I had to google this 😂 [link](https://en.wikipedia.org/wiki/Fully_qualified_domain_name)

So would it be accurate to say, "In order to enable https, you have to add an external domain." I'm thinking we could add something like that to the docs.

**Reply by aflextr** [reply]
@jsjoeio Yes, you need to attach it to the documentation. The fqdn event was available in aws,oracle and azure so it had the ability to connect directly as https without any hassle. This feature worked best in Azure.
 Well done azure :D .In the past there was no code server error as http but now the program doesn't work properly without it. Do you have a roadmap to fix this?

**Reply by jsjoeio** [reply]
Hmm...I mean I think that has to be a change in the browser (complaining about http vs. https), right? So I'm not sure if we can fix this, but I could be misunderstanding.

What browser are you using?

**Reply by jsjoeio** [reply]
Made an issue here: https://github.com/cdr/code-server/issues/3921

Feel free to add thoughts there!

**Reply by aflextr** [reply]
> Hmm...I mean I think that has to be a change in the browser (complaining about http vs. https), right? So I'm not sure if we can fix this, but I could be misunderstanding.
> 
> What browser are you using?

Yeah, I am using Google Chrome. Same bug was present in other chromium based browsers but now almost all browsers are affected.

**Reply by lg-nineoneninethreenine** [reply]
What about Firefox? Or Opera?

**Comment by lg-nineoneninethreenine** [comment]
> How can I make my virtual machine https domain in google cloud?
> 
> 
> Note: "code-server --link" not working.

Not an answer by the way. `code-server --link` works best of you have a `.cdr.co` domain from Coder. I haven't tried it but have signed up for it. I must figure out how to login though.

**Reply by jsjoeio** [reply]
When running `code-server --link`, we serve up code-server for you. All you have to do is login with GitHub and you'll be good to go! See [this video](https://www.youtube.com/watch?v=MdIEiZpg5vQ)
